[FE / 회원가입] 회원가입 뷰, 로그인 뷰

ummaeha opened the pull request:
2022-01-18 회원가입 뷰 데스크탐 뷰only,,,,

Before merging...

**뷰**

- [x] 회원가입 데스크탑 뷰
- [x] 로그인 데스크탑 뷰
- [ ] 반응형처리

**프론트쪽 체크 로직** 
- [x] 이메일 형식, 비밀번호 형식 validation

**통신 - 요청** 
- [x] 이메일 중복확인 POST `/auth/email`
- [x] 회원가입 요청 POST `/auth/signup`
- [x] 로그인 요청 POST `/auth/login`

**그 외(TBD - 논의 필요)**
: 마이너한 기능으로 추후 논의 하기로 함
- 비밀번호 변경
- 회원 탈퇴
-  로그아웃




![image](https://user-images.githubusercontent.com/48500209/150053067-b661e4cf-108d-482b-a3f7-924bf7c42033.png)
![image](https://user-images.githubusercontent.com/48500209/149978136-544449ed-03c7-42be-a8c0-2fbef75a9910.png)

ummaeha committed:
feat: 회원가입 뷰 데스크탑뷰 Only(반응형 x)

ummaeha committed:
feat: 로그인 페이지 뷰((반응형 x)

ummaeha commented:
@minjyo  지호님! navbar 접히는거 들어가면 혹시 깨질까 싶어서 일단 데스크탑 뷰만 만들었는데 nav바 fixed로 들어가있으면 상관없으려나요?

ummaeha committed:
chore: 중복확인 버튼 추가

minjyo commented:
네 괜찮을 것 같습니다

ummaeha committed:
cchore: 회원가입 폼 디자인 수정

ummaeha committed:
feat: email, password 체크하는 프론트 로직 추가
전체 필드의 유효성을 체크하는 로직은 checkAll 함수에 작성될 예정이며, 인증 백엔드와 response에 대해 협의하고 진행할 예정

ummaeha committed:
chore: console.log삭제, optional 항목 error 잡기

ummaeha committed:
Merge branch 'develop' of https://github.com/STOVE-Milk/steam-clone into feature/front-auth

ummaeha committed:
chore:  merge conflict  해결, layout padding 없앰

minjyo committed:
chore: 네브바 수정된거 가져오기, redux store와 getServersideProps간 에러 해결 가져오기

ummaeha committed:
chore: user reducer 작성을 위한 파일 생성

ummaeha committed:
chore: env파일에서 따옴표 삭제
env에서 localhost:8080 을 가져갈때 따옴표까지 함께 가져가서 api 작성하는 파일에서 url 이 이상하게 들어가는 이슈 해결

ummaeha committed:
chore: category 페이지 useEffect -> serverside props로 변경

ummaeha committed:
feat: 회원가입 validation 로직 파일 분리 및 signup 페이지 수정

ummaeha committed:
chore: navbar때문에꼬여버린  😵 깃 정리중,,,

ummaeha committed:
chore: navbar 문제 해결 중

ummaeha committed:
chore: merge conflict 처리

ummaeha committed:
feat: 이메일중복확인 API요청

ummaeha committed:
feat: 회원가입 요청(by dispatch)

ummaeha committed:
chore: merge conflict 처리

ummaeha committed:
feat: 로그인 요청(/auth/login)

minjyo commented on src/frontend/components/molecules/AuthInput.tsx:
```diff
@@ -0,0 +1,64 @@
+import styled from 'styled-components';
+import Text from 'components/atoms/Text';
+import FilledButton from 'components/atoms/FilledButton';
+export interface IAuthInputProps {
```
지엽적이긴 한데 인터페이스 위 아래로 엔터..!!

minjyo commented on src/frontend/components/molecules/AuthInput.tsx:
```diff
@@ -0,0 +1,64 @@
+`;
+
+export default function AuthInput({
+  title,
+  type,
+  placeholder,
+  name,
+  onChange,
+  checkValidation,
+  warningMsg,
+}: IAuthInputProps) {
```
이부분 props: IAuthInputProps로 쓰는거랑 같은거 아시죠...!? 양하님처럼 쓰면 밑에서 props.를 안써도 되는 이점도 있구요

ummaeha replied:
아 네네 알고있습니다/ 원래 저렇게 props가 많아질지 모르고 {}로 풀어서쓴건데 밑에 코드들도 다 props. 으로 안 쓰고 풀어서 쓴 상태라 그대로 뒀네요 ~

minjyo commented on src/frontend/components/molecules/AuthSelectBox.tsx:
```diff
@@ -0,0 +1,43 @@
+  }
+`;
+
+export default function AuthSelectBox({ title, option, onChange }: IOptionProp) {
```
나중에 리팩토링 할때 이전에 만들어뒀던 atom의 SelectBox를 재사용해서 코드를 작성해보는 것도 좋을것같아요

minjyo commented on src/frontend/pages/signin.tsx:
```diff
@@ -0,0 +1,75 @@
+    email: '',
+    password: '',
+  });
+  const [errors, setErrors] = useState({} as Record<any, string>);
```
string이 아닌 Record<>로 정의하신 이유가 있나요?

ummaeha replied:
전에 참여했던 프로젝트에서 썻던대로 가져왔는데, setError라는 error를 set하는 함수의 타입을 임의대로 key value로 지정하기위해 사용했다고 이해하고 적었습니다. 여기서 사용된 사례들은 다 key값이 string으로 들어가있긴하지만, 확장되면 오류메시지는 항상 string일테지만, key에는 다른 것이 들어갈 수 있다고 생각했습니다.

ummaeha replied:
지금은 Key가 다 string이라서 Record<string, string>으로 해도 문제는 없을 것 같습니다 :)

minjyo commented on src/frontend/pages/signup.tsx:
```diff
@@ -1,7 +1,179 @@
+
+  const checkEmail = (email: string) => {
+    if (email.length === 0) {
+      setErrors((prev) => ({ ...prev, email: '이메일을 필수로 입력해주세요' }));
```
이부분 이해가 잘 안가는데 내일 설명부탁드립니당

ummaeha replied:
이전에 있던 다른쪽에서 set된 에러메시지(key, value)를 보관하기위해 ...prev를 해준것이고 새로추가된 메시지를 뒤에 더한 것 입니다. 이 부분에선는 email에 대한 에러가 발생할 예정이므로, email에러를 추가한 것 입니다.

ummaeha committed:
chore: 리뷰 반영
- 콘솔로그 삭제
- 주석 더하기
	- [refer]: 참고한 코드가 있는 경우
	- [explain]: 설명이 필요하다고 생각하는 부분
	- TO DO: 앞으로 해야할 일

ummaeha committed:
chore: 인증서버 base url설정

ummaeha committed:
Merge branch 'develop' of https://github.com/STOVE-Milk/steam-clone into feature/front-auth

ummaeha committed:
feat: signup, signin 통신 테스트 완료, jwt 토큰 복호화 완료, to do : 유저정보 store에 넣기

ummaeha committed:
feat: userInfo를 store에 저장, signup, signin 기능 디테일

ummaeha committed:
chore: 로그인 요청시 중복체크 여부 확인, alert메시지 object 나오는거 수정

ummaeha committed:
chore: 커밋 누락 내용 추가

ummaeha commented:
리뷰때 중점적으로 봐주셨으면 좋겠는 부분은 `signup.tsx` 파일 입니다. 
회원가입을 위한 조건들때문에` if `문을 많이 걸어놓았는데, 그 부분을 어떻게 하면 개선할 수 있을지 아이디어 주시면 감사하곘습니다:)

ummaeha commented:
@minjyo 
어제 이야기한대로, private routeing 처리는 후순위로 하고, 일단 각 페이지에서 useSelector로 정보를 가져오는 것으로 하겠습니다 :)

minjyo commented on src/frontend/pages/signin.tsx:
```diff
@@ -0,0 +1,108 @@
+      alert(res.message);
+      if (res.code === 10000) {
+        //성공
+        localStorage.setItem('accessToken', res.data.accessToken);
+        localStorage.setItem('refreshToken', res.data.refreshToken);
+        //[refer]: jwt decode하는 코드 | 출처: https://archijude.tistory.com/432
+        const token = res.data.accessToken;
+        const base64Payload = token.split('.')[1]; //value 0 -> header, 1 -> payload, 2 -> VERIFY SIGNATURE
+        const payload = Buffer.from(base64Payload, 'base64');
+        const result = JSON.parse(payload.toString());
```
토큰 파싱하는 부분은 따로 함수로 빼면 깔끔할 것 같아요

ummaeha replied:
좋은 생각입니다!

minjyo commented on src/frontend/pages/signin.tsx:
```diff
@@ -0,0 +1,108 @@
+        ...prev,
+        nullChecker: '빈칸을 채워주세요',
+      }));
+      alert(Object.values(errors));
```
시간 되실 때 alert한 부분 전부 모달로 사용하셔도 될 것 같습니다.

ummaeha replied:
넵 이부분도 리팩토링때 한 번에 진행할게요 ! 지금 제쪽에서 안내메시지를 alert처리해놓은 곳이 은근 많아서요!

minjyo commented on src/frontend/pages/signin.tsx:
```diff
@@ -0,0 +1,108 @@
+      }
+    }
+  };
+  const onChangeSetInfo = (e: any) => {
```
React.memo !

ummaeha replied:
React.memo를 씌워봤는데, 콘솔을 찍어봤을때는 똑같이 onChange이벤트에 따라 렌더링이 되어서, 좀 더 찾아보고 리팩토링때 적용하도록 하겠습니다 :) 노션에 메모해두겠습니다!
![image](https://user-images.githubusercontent.com/48500209/152992516-f0faf05f-bb72-4039-820f-30c2b6f85291.png)

minjyo commented on src/frontend/pages/signup.tsx:
```diff
@@ -1,7 +1,245 @@
+      each.length == 0 && (nullChecker = false);
+    });
+
+    if (!nullChecker) {
```
회원가입, 로그인할 때 유효값인지 검증하는 로직 자체가 지엽적이라 if문 관련 리팩토링은 힘들거 같은데, 
현재 시점에선 null check하는 부분과 오류 관련 모달(alert를 모달로 바꾼다면)을 유틸로 뺼 수 있을 것 같습니다. 
회원가입, 로그인화면에서 동일하게 쓰이니까요

ummaeha replied:
조언 감사합니다. 제가 setState함수를 어떻게 util로 뺄지 좀 더 고민해보고 리팩토링때 적용해보도록하겠습니다.

ummaeha committed:
feat: JWT토큰 파싱 util화
